tests/api/bloom.js

```diff
@@ -0,0 +1,76 @@
+const tape = require('tape')
+const Bloom = require('../../lib/bloom')
+const utils = require('ethereumjs-util')
+
+const byteSize = 256
+
+tape('bloom', (t) => {
+  t.test('should initialize without params', (st) => {
+    const b = new Bloom()
+    st.deepEqual(b.bitvector, utils.zeros(byteSize), 'should be empty')
+    st.end()
+  })
+
+  t.test('shouldnt initialize with invalid bitvector', (st) => {
+    st.throws(() => new Bloom('invalid'), /AssertionError/, 'should fail for invalid type')
+    st.throws(() => new Bloom(utils.zeros(byteSize / 2), /AssertionError/), 'should fail for invalid length')
+    st.end()
+  })
+
+  t.test('should contain values of hardcoded bitvector', (st) => {
+    const hex = '00000000000000000000000000000000000000000000000000000000000000000100000000000000000000000000010000000000000000000000000000000000000000000000000000000000000000008000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000200000020000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000002000000000000000000000000000000000000000000000000000000000000000000'
+    const vector = Buffer.from(hex, 'hex')
+
+    const b = new Bloom(vector)
+    st.true(b.check('value 1'), 'should contain value 1')
+    st.true(b.check('value 2'), 'should contain value 2')
+    st.end()
+  })
```

Comment:
I actually am no bloom filter expert myself, could you explain - just for my understanding - where this 'value ' part of 'value 1' can be re-found in the hex string?

Reply:
Back then I also looked at a helpful [tutorial](https://llimllib.github.io/bloomfilter-tutorial/) to learn their details, which mentions "To add an element to the Bloom filter, we simply hash it a few times and set the bits in the bit vector at the index of those hashes to 1."